src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptionsExtensions.cs

```diff
@@ -43,7 +44,41 @@ public static Channel CreateChannel(this OtlpExporterOptions options)
             }
 
 #if NETSTANDARD2_1 || NET6_0_OR_GREATER
-            return GrpcChannel.ForAddress(options.Endpoint);
+            // The specification does not dictate the parameters of the default retry policy.
+            // These defaults are borrowed from opentelemetry-java.
+            var retryPolicy = new RetryPolicy
+            {
+                // https://github.com/open-telemetry/opentelemetry-java/blob/6f755cc8128f8077643b4990327c22b293c46e45/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/retry/RetryPolicyBuilder.java#L19-L22
+                MaxAttempts = 5,
+                InitialBackoff = TimeSpan.FromSeconds(1),
+                MaxBackoff = TimeSpan.FromSeconds(5),
+                BackoffMultiplier = 1.5,
+
+                // https://github.com/open-telemetry/opentelemetry-java/commit/6f755cc8128f8077643b4990327c22b293c46e45#diff-2d25b543bc2c5a13e495895b6f6a2db80e1167bbe3387024cb917efda35a99acR24-R34
+                RetryableStatusCodes =
+                {
+                    StatusCode.Cancelled,
+                    StatusCode.DeadlineExceeded,
+                    StatusCode.ResourceExhausted,
```

Comment:
TODO: add a test validating that the Grpc.Net.Client library honors the `retry_delay` sent from the server.

Reply:
The [server ](https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures:~:text=To%20signal%20backpressure%20when%20using%20gRPC%20transport%2C%20the%20server%20MUST%20return%20an%20error%20with%20code%20Unavailable%20and%20MAY%20supply%20additional%20details%20via%20status%20using%20RetryInfo.)could be sending a `retry_delay` even with `StatusCode.Unavailable`. Client should be respecting that as well?

Reply:
@JamesNK I'm curious if you have a moment to share guidance here...

This PR implements retry for our gRPC exporter using a `RetryPolicy`. There are a number of status codes which can be retried, however, we have a [requirement](https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#failures) that cannot be expressed using a `RetryPolicy` alone when status is `Unavailable` or `Resource_Exhausted`:

> The server MUST indicate retryable errors using code [Unavailable](https://godoc.org/google.golang.org/grpc/codes) and MAY supply additional [details via status](https://godoc.org/google.golang.org/grpc/status#Status.WithDetails) using [RetryInfo](https://github.com/googleapis/googleapis/blob/6a8c7914d1b79bd832b5157a09a9332e8cbd16d4/google/rpc/error_details.proto#L40) containing 0 value of RetryDelay.

My hope was to avoid rolling our own retry mechanism, but from what I can tell Grpc.Net.Client does not support handling [RetryInfo](https://github.com/googleapis/googleapis/blob/6a8c7914d1b79bd832b5157a09a9332e8cbd16d4/google/rpc/error_details.proto#L40).

Options I can think of
1. Roll our own retry mechanism for all status codes conforming to our requirements.
2. Use a RetryPolicy for retryable codes which do require handling RetryInfo and only roll our own logic for Unavailable and Resource_Exhaused.

Are there benefits to one option over the other? Are there better options?

Reply:
> Do other gRPC client libraries support RetryInfo?

No, not from what I've seen. This RetryInfo seems to be something specific to some Google APIs. It's unfortunate that the OTel spec adopted this over the standard gRPC retry spec.

OTLP supports both gRPC and HTTP and since HTTP has a [`Retry-After`](https://www.rfc-editor.org/rfc/rfc9110.html#section-10.2.3) response header I suspect the decision to adopt this RetryInfo thing for gRPC was inspired by the desire to keep parity with the retry behavior between both HTTP and gRPC.

Reply:
gRPC retry supports the [`grpc-retry-pushback-ms`](https://github.com/grpc/proposal/blob/master/A6-client-retries.md#pushback) response header which seems similar.

Grpc.Net.Client retries doesn't support RetryInfo. I suggest you look at other implementations to see what they do. If they have a clever technique then maybe that is something that could also be done in .NET. If they are all doing retries manually then you'll have to do it manually.